ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_submission_service.proto

```diff
@@ -37,11 +37,17 @@ service CommandSubmissionService {
   // Errors:
   // - ``UNAUTHENTICATED``: if the request does not include a valid access token
   // - ``PERMISSION_DENIED``: if the claims in the token are insufficient to perform a given operation
-  // - ``NOT_FOUND``: if the request does not include a valid ledger id
+  // - ``NOT_FOUND``: if the request does not include a valid ledger id or the submission rank or
+  //   deduplication period are too old.
   // - ``INVALID_ARGUMENT``: if the payload is malformed or is missing required fields
   // - ``UNAVAILABLE``: if the participant is not yet ready to submit commands or if the service has been shut down.
   // - ``RESOURCE_EXHAUSTED``: if the participant or the ledger is overloaded. Clients should back off exponentially and retry.
-  // - ``ABORTED``: if a contract key is missing or duplicated due to for example contention on resources
+  // - ``ABORTED``: if a contract key is missing or duplicated due to for example contention on resources or
```

Comment:
@andreaslochbihler-da : is the contract-key case really a synchronous error? I don't have a good model of where and how that happens.

Reply:
It can be synchronous or asynchronous, depending on where the check fails (during submission on the participant or during conflict detection on the ledger).